src/tests/ftest/recovery/pool_cleanup.py

```diff
@@ -0,0 +1,157 @@
+        # Add -fault to the label and use that to call dmg pool get-prop.
+        orig_identifier = pool.identifier
+        pool.label.update(orig_identifier + "-fault")
+        ps_label = pool.get_property(prop_name="label")
+        if ps_label != orig_identifier:
+            msg = (f"Unexpected label in PS! Expected = {orig_identifier}; Actual = "
+                   f"{ps_label}")
+            errors.append(msg)
+        # Restore the label to the original.
+        pool.label.update(orig_identifier)
+
+        # 5. Stop the servers and enable the checker.
+        self.log_step("Stop the servers and enable the checker.")
+        dmg_command.check_enable()
+
+        # 6. Set the policy to --all-interactive.
+        self.log_step("Set the policy to --all-interactive.")
+        dmg_command.check_set_policy(all_interactive=True)
+
+        # 7. Start the checker and query the checker until the fault is detected.
+        self.log_step("Start and query the checker until the fault is detected.")
+        seq_num = None
+        # Start checker.
+        dmg_command.check_start()
+        # Query the checker until expected number of inconsistencies are repaired.
+        for _ in range(8):
+            check_query_out = dmg_command.check_query()
+            # Status is INIT before starting the checker.
+            if check_query_out["response"]["status"] == "RUNNING":
+                seq_num = check_query_out["response"]["reports"][0]["seq"]
+                break
+            time.sleep(5)
+        if not seq_num:
+            self.fail("Checker didn't detect any fault!")
+
+        # 8. Repair with option 1; Trust PS pool label.
+        self.log_step("Repair with option 1; Trust PS pool label (in JSON output).")
+        dmg_command.check_repair(seq_num=seq_num, action=1)
+
+        # 9. Query the checker until the fault is repaired.
+        self.log_step("Query the checker until the fault is repaired.")
+        repair_report = self.wait_for_check_complete()[0]
+
+        # Verify that the repair report has expected message "Update the MS label".
+        action_message = repair_report["act_msgs"][0]
+        exp_msg = "Update the MS label"
+        errors = []
+        if exp_msg not in action_message:
+            errors.append(f"{exp_msg} not in {action_message}!")
+
+        # 10. Call dmg check disable and restart the servers.
+        self.log_step("Call dmg check disable and restart the servers.")
+        dmg_command.check_disable()
+
+        # 11. Wait for all ranks to join.
+        self.log_step("Wait for all ranks to join.")
+        rank_list = self.server_managers[0].get_host_ranks(hosts=self.hostlist_servers)
+        failed_ranks = self.server_managers[0].check_rank_state(
+            ranks=rank_list, valid_states="joined", max_checks=10)
+        if failed_ranks:
+            self.fail(
+                f"Following ranks didn't become joined after restart! {failed_ranks}")
```

Comment:
Should this use `errors.append` to match above? What if there were already errors before we get here?

Reply:
Done